### ISSUE TITLE
Non-unique name of struct Ex__1 causes a compilation error.

### ISSUE THREAD

**Issue by Oleg-N-Cher** [jtempl/ofront#43]
**GameHunter** reports this bug:

```
MODULE Ex1;

TYPE
  Rec0 * = RECORD
    y * : RECORD
    END;
  END;

END Ex1.
```

```
MODULE Ex;

IMPORT Ex1;

TYPE
  Rec1 * = RECORD ( Ex1.Rec0 )
  END;
  Rec2 * = RECORD ( Ex1.Rec0 )
  END;

END Ex.
```

```
/*  Ofront 1.2 -xtpkaem */
#include "SYSTEM.h"
#include "Ex1.h"

typedef
	struct Ex_Rec1 { /* Ex1_Rec0 */
		struct Ex__1 {
			char _prvt0[1];
		} y;
	} Ex_Rec1;

typedef
	struct Ex_Rec2 { /* Ex1_Rec0 */
		struct Ex__1 {
			char _prvt0[1];
		} y;
	} Ex_Rec2;
```

![image](https://user-images.githubusercontent.com/1090608/130973460-f521a8e4-a151-42b7-a79f-6393d96cd9ad.png)

**Comment by Oleg-N-Cher** [comment]
CPfront (Ofront modified by OMinc) shows a slightly more tricky description of the structure (see the parameters of Ex_IncRec_Inc):

```
#include "SYSTEM.h"

static INTEGER Ex_i;
static struct Ex_IncRec1 {
	char _prvt0[1];
} Ex_r;

export INTEGER Ex_IncRec__desc[];
export SYSTEM_TYPEDESC *Ex_IncRec__typ = (SYSTEM_TYPEDESC*)(Ex_IncRec__desc + 3);
export INTEGER Ex_IncRec1__desc[];
export SYSTEM_TYPEDESC *Ex_IncRec1__typ = (SYSTEM_TYPEDESC*)(Ex_IncRec1__desc + 3);

static void Ex_IncProc (struct Ex_IncRec {
	char _prvt0[1];
} *r_param, SYSTEM_TYPEDESC *r_param__typ, INTEGER *i);

export void Ex__reg();
export void Ex__body();
export struct SYSTEM_MODDESC Ex__desc;


static void Ex_IncRec_Inc (struct Ex_IncRec {
	char _prvt0[1];
} *r, SYSTEM_TYPEDESC *r__typ, INTEGER *i)
{
	__ENTER("Ex.IncRec.Inc");
	*i = *i + 1;
	__EXIT;
}
```

**Comment by jtempl** [comment]
Dear Oleg,

in hindsight I am not sure that hiding the private members of a RECORD are
a good idea at all. It makes debugging harder and complicates the
translator. A simpler approach may have been to prefix private members with
an _ or similar and export all related internal stuff in the header file.
But it may also lead to other problems...

Nevertheless it should be possible to fix this bug by properly incrementing
some internal counter for naming the struct correctly.

- Josef

Oleg N. Cher ***@***.***> schrieb am Do., 26. Aug. 2021,
15:44:

**Comment by jtempl** [comment]
maybe it suffices to avoid the definition of the struct after it has been
defined once leading to


typedef
	struct Ex_Rec2 { /* Ex1_Rec0 */
		struct Ex__1 y;
	} Ex_Rec2;


- Josef



Oleg N. Cher ***@***.***> schrieb am Do., 26. Aug. 2021,
15:44:

**Comment by Oleg-N-Cher** [comment]
Dear Josef,
Thank you for your quick response.
Recently, I have been turning to you only in the most difficult cases, when I myself am lost in uncertainty.
This is just such a case. And I certainly tend to borrow the solution from CPfront, since it's easier for me to do this than anything else.
But I'm interested in how you will do it.
Apparently, the second way will be more acceptable than increasing the counter. I'm not sure, though.

**Comment by Oleg-N-Cher** [comment]
I looked at this error more carefully and I can say that it is not solved in CPfront either.

![image](https://user-images.githubusercontent.com/1090608/131248210-87ab52aa-2943-4259-bb02-d5b4e7826da2.png)

Thus, we do not have a ready-made solution for this problem. I will try to connect Dmitry Dagaev, the developer of Multi-Oberon (partly based on CPfront).

**Comment by Oleg-N-Cher** [comment]
Dmitry Dagaev suggests that instead of duplicating the internal structure Ex1__1, there should be the use of anonymous structure defined in the base module Ex1. What do you think, Josef?

**Comment by Oleg-N-Cher** [comment]
Yeah, it compiles! (I removed the name Ex1__1 and made the internal struct as anonymous).
```
/* Ofront+ 1.0 -mxtC -48 */
#include "SYSTEM.oh"
#include "Ex1.oh"

typedef
	struct Ex_Rec1 { /* Ex1_Rec0 */
		struct {
			char _prvt0[1];
		} y;
	} Ex_Rec1;

typedef
	struct Ex_Rec2 { /* Ex1_Rec0 */
		struct {
			char _prvt0[1];
		} y;
	} Ex_Rec2;
```
But probably, this is not exactly what Mr. Dagaev suggests.

**Comment by Oleg-N-Cher** [comment]
I suggest this solution. In OPC.DeclareBase:

![image](https://user-images.githubusercontent.com/1090608/137461644-262e941b-4029-44ce-a377-d138b380e9d8.png)

The generated code was:

```
typedef
	struct Ex_Rec2 { /* Ex1_Rec0 */
		struct Ex__1 {
			char _prvt0[1];
		} y;
	} Ex_Rec2;
```

Now:

```
typedef
	struct Ex_Rec2 { /* Ex1_Rec0 */
		struct /* Ex__1 */ {
			char _prvt0[1];
		} y;
	} Ex_Rec2;
```

I checked compilability in GCC, Clang, Turbo C v2.01, cc65, SDCC and SCCZ80.
